source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,196 @@
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    PlatenessStatistics<dim>::execute (TableHandler &statistics)
+    {
+      // Create a quadrature formula based on the velocity element.
+      const types::boundary_id top_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+
+      const unsigned int quadrature_degree = this->introspection().polynomial_degree.temperature;
+      const QGauss<2> quadrature_formula_face(quadrature_degree);
+
+      FEFaceValues<3> fe_face_values (this->get_mapping(),
+                                      this->get_fe(),
+                                      quadrature_formula_face,
+                                      update_values |
+                                      update_quadrature_points |
+                                      update_JxW_values);
+
+      // Vector to store the integral of the second invariant of the strain rate
+      // over the surface.
+      std::vector<double> local_second_invariant_of_strain_rate_integral;
+      std::vector<double> local_surface_area_integral;
+      std::vector<double> local_second_invariant_of_strain_rate_and_area;
+      std::vector<double> sorted_local_second_invariant_of_strain_rate;
+
+      const unsigned int n_q_points = quadrature_formula_face.size();
+      const unsigned int n_compositional_fields = this->n_compositional_fields();
+      //std::vector<double> compositional_values(n_q_points);
+
+      typename MaterialModel::Interface<dim>::MaterialModelInputs in(n_q_points,
+                                                                     n_compositional_fields);
+      typename MaterialModel::Interface<dim>::MaterialModelOutputs out(n_q_points,
+                                                                       n_compositional_fields);
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned() && cell->at_boundary())
+          {
+            unsigned int face_idx = numbers::invalid_unsigned_int;
+            bool at_upper_surface = false;
+            {
+              for (unsigned int f=0; f<GeometryInfo<3>::faces_per_cell; ++f)
+                {
+                  if (cell->at_boundary(f) && cell->face(f)->boundary_id() == top_boundary_id)
+                    {
+                      // If the cell is at the top boundary, assign face_idx.
+                      face_idx = f;
+                      at_upper_surface = true;
+                      break;
+                    }
+                }
+
+            }
+            fe_face_values.reinit(cell, face_idx);
+
+            if (at_upper_surface)
+              {
+                in.reinit(fe_face_values,
+                          cell,
+                          this->introspection(),
+                          this->get_solution());
+                //in.requested_properties = MaterialModel::MaterialProperties::viscosity;
```

Comment:
What is this line about?

Reply:
not needed, since we do not need the MaterialModel at all.